### PR TITLE
SDK-178: Adding File & Partition Sensors

### PR DESCRIPTION
**Pull request by msumit**
Note: Sensors api will be in prod with rb41.

### CODE REVIEW

**Comment by yogesh2021** [comment]
A qubole operator change will also follow up with this ?

**Comment by msumit** [comment]
@yogesh2021 yes there is PR opened in Airflow repo (bitbucket) as well..

File: qds_sdk/sensors.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        Method to call check after parsing args from cli
+        :param args: inline arguments
+        :return: True or False
+        """
+        parser = cls.parsers()
+        parsed = parser.parse_args(args)
+        return cls.check(json.loads(parsed.data))
+
+    @classmethod
+    def parsers(cls):
```
Comment (harshshah87) [inline qds_sdk/sensors.py]: The function parsers and check_cli(we can rename it to check) should be in new SensorCmdLine class. The dependency on usage and description from the derived classes can be removed.
usage = "qds.py %s check -d 'json string'" % cls.__name__.lower()
description (do we really need one) can be set in similar manner.

Comment (msumit) [inline qds_sdk/sensors.py]: Any specific reason to move these methods into new `SensorCmdLine` class? I followed the design of `Command` class here.. Also I think `cls` variable won't be available in parent class, as its not inside a class method?

Comment (msumit) [inline qds_sdk/sensors.py]: @harshshah87 moved `check` and `parser` to `SesnorCmdLine` class..

File: qds_sdk/sensors.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import logging
+import json
+
+log = logging.getLogger("qds_sensors")
+
+
+class SensorCmdLine:
+
+    @staticmethod
+    def check(cls, args):
```
Comment (harshshah87) [inline qds_sdk/sensors.py]: NIT: Need to rename the variable from cls to sensor_class. The \@staticmethod along with cls as variable name is confusing.

**Comment by msumit** [comment]
@harshshah87 done.. can u plz squash and merge?

**Comment by harshshah87** [comment]
Merged to unreleased.

**Pull request closed without merging**